File: app/src/main/java/com/booboot/vndbandroid/bean/vndbandroid/Priority.java
package com.booboot.vndbandroid.bean.vndbandroid;

/**
 * Created by od on 15/03/2016.
 */
public class Priority {
    public final static int HIGH = 0;
    public final static int MEDIUM = 1;
    public final static int LOW = 2;
    public final static int BLACKLIST = 3;
    public final static String DEFAULT = "Add to my wishlist";

    public static String toString(int priority) {
        switch (priority) {
            case 0:
                return "High";
            case 1:
                return "Medium";
            case 2:
                return "Low";
            case 3:
                return "Blacklist";
            default:
                return DEFAULT;
        }
    }

    public static String toShortString(int priority) {
        switch (priority) {
            case 0:
                return "H";
            case 1:
                return "M";
            case 2:
                return "L";
            case 3:
                return "B";
            default:
                return "-";
        }    }
}

File: app/src/main/java/com/booboot/vndbandroid/model/vndbandroid/Priority.kt
package com.booboot.vndbandroid.model.vndbandroid

import com.booboot.vndbandroid.App
import com.booboot.vndbandroid.R

object Priority {
    const val HIGH = 0
    const val MEDIUM = 1
    const val LOW = 2
    const val BLACKLIST = 3
    const val DEFAULT = "Add to my wishlist"
    val ALL = listOf(HIGH, MEDIUM, LOW, BLACKLIST)

    fun toString(priority: Int?) = when (priority) {
        0 -> "High"
        1 -> "Medium"
        2 -> "Low"
        3 -> "Blacklist"
        else -> null
    }

    fun toShortString(priority: Int?): String = when (priority) {
        0 -> "H"
        1 -> "M"
        2 -> "L"
        3 -> "B"
        else -> App.context.getString(R.string.dash)
    }
}